memory_partition.go

```diff
@@ -236,8 +236,8 @@ func (m *memoryMetric) selectPoints(start, end int64) []*DataPoint {
 	} else {
 		// Use binary search because points are in-order.
 		endIdx = sort.Search(int(size), func(i int) bool {
-			return m.points[i].Timestamp < end
-		}) + 1
+			return m.points[i].Timestamp > end
```

Comment:
In order to prevent the result from containing a point where the timestamp is exact the same as the `end`, it's better to use `>=`.

```suggestion
			return m.points[i].Timestamp >= end
```

Reply:
I've changed that and squashed the commits